client/components/d3-base/index.js

```diff
@@ -41,6 +41,10 @@ export default class D3Base extends Component {
 	 * Determines if new props were provided, and updates the local state accordingly. This basically stores any new
 	 * prop in the local state (so it can be used for comparison in future calls), and resets the list of params in that
 	 * case (in order to indicate that the chart should be drawn again).
+	 *
+	 * @param {Object} nextProps the next props
+	 * @param {Object} prevState the previous state
+	 * @returns {Object} the next state
```

Comment:
I would opt for replacing `/**` with `/*` in the comment to disable the JSDoc warning. It's better than adding worthless docs of a React API method just to satisfy the linter.

Reply:
i like that idea. I've been debating turning off jsdoc errors entirely..